openpype/plugins/publish/extract_slate_global.py

```diff
@@ -0,0 +1,538 @@
+    using headless chrome/chromium. They then get converted using oiio,
+    with an ffmpeg fallback (still not implemented).
+
+    It will work only on represenations having `slateGlobal = True` and
+    `tags` including `slate`
+    """
+
+    label = "Extract Slate Global"
+    order = pyblish.api.ExtractorOrder + 0.0305
+    families = ["slate"]
+    hosts = [
+        "nuke",
+        "maya",
+        "blender",
+        "houdini",
+        "shell",
+        "hiero",
+        "premiere",
+        "harmony",
+        "traypublisher",
+        "standalonepublisher",
+        "fusion",
+        "tvpaint",
+        "resolve",
+        "webpublisher",
+        "aftereffects",
+        "flame",
+        "unreal"
+    ]
+
+    _slate_data_name = "slateGlobal"
+
+    def process(self, instance):
+
+        if self._slate_data_name not in instance.data:
+            self.log.warning(
+                "Slate Global workflow is not active, skipping slate extraction..."
+            )
+            return
+
+        if "representations" not in instance.data:
+            self.log.error("No representations to work on!")
+            raise ValueError("No items in list.")
+
+        repre_ignore_list = ["thumbnail", "passing"]
+
+        slate_data = instance.data[self._slate_data_name]
+
+        # get pyblish comment and intent
+        common_data = slate_data["slate_common_data"]
+        common_data["comment"] = instance.context.data.get("comment") or "-"
+        intent = instance.context.data.get("intent")
+        if not isinstance(intent, dict):
+            intent = {
+                "label": intent,
+                "value": intent
+            }
+        common_data["intent"] = intent
+
+        # init SlateCreator Object that creates the slate frames
+        slate_creator = SlateCreator(
+            template_path=slate_data["slate_template_path"],
+            resources_path=slate_data["slate_resources_path"],
+            data=common_data,
+        )
+
+        # loop through representations to find thumbnail path
+        repre_thumbnail_path = ""
+        for thumb_repre in instance.data["representations"]:
+            if thumb_repre["name"] == "thumbnail":
+                repre_thumbnail_path = os.path.join(
+                    thumb_repre["stagingDir"], thumb_repre["files"]
+                )
+
+        repre_match = None
+
+        # loop through representations and generate a slate frame for each
+        for repre in instance.data["representations"]:
+            if repre["name"] in repre_ignore_list:
+                self.log.debug(
+                    "Representation '{}' was ignored.".format(repre["name"])
+                )
+                continue
+
+            # check if repre is a sequence
+            is_sequence = False
+            check_file = repre["files"]
+            if isinstance(check_file, list):
+                check_file = check_file[0]
+                is_sequence = True
```

Comment:
Is this more readable?
```suggestion
            # check if repre is a sequence
            is_sequence = isinstance(check_file, list)
            check_file = repre["files"][0] if is_sequence else repre["files"]
```

Reply:
yes! I like it, changing it

Reply:
oops, `check_file` isn't defined otherwise, changed to:
```python
is_sequence = isinstance(repre["files"], list)
check_file = repre["files"][0] if is_sequence else repre["files"]
```